src/content/app/genome-browser/state/browser-bookmarks/browserBookmarksSlice.ts

```diff
@@ -113,85 +93,59 @@ export const updatePreviouslyViewedObjectsAndSave =
         ? [geneSymbol, stable_id as string]
         : activeFocusObject.label;
 
-    const newObject = {
+    const newObject: PreviouslyViewedObject = {
       genome_id: activeFocusObject.genome_id,
       object_id: activeFocusObject.object_id,
       type: activeFocusObject.type,
       label: label
     };
 
-    const updatedEntitiesArray = [newObject, ...previouslyViewedObjects];
+    const updatedObjects = [newObject, ...previouslyViewedObjects];
 
     // Limit the total number of previously viewed objects to 250
-    const previouslyViewedObjectsSlice = updatedEntitiesArray.slice(-250);
+    const previouslyViewedObjectsSlice = updatedObjects.slice(-250);
 
     browserBookmarksStorageService.updatePreviouslyViewedObjects({
       [activeGenomeId]: previouslyViewedObjectsSlice
     });
 
-    const data = {
-      ...state.browser.browserBookmarks[activeGenomeId],
-      previouslyViewedObjects: previouslyViewedObjectsSlice
-    };
-    const persistentBrowserBookmarksProperties =
-      pickPersistentBrowserBookmarksProperties(data, activeGenomeId);
-
-    browserBookmarksStorageService.updateBookmarks({
-      [activeGenomeId]: persistentBrowserBookmarksProperties
-    });
-
     dispatch(
-      updateBrowserBookmarksForGenome({
+      bookmarksSlice.actions.updatePreviouslyViewedObjects({
         activeGenomeId,
-        data
+        updatedObjects
```

Comment:
Not `previouslyViewedObjectsSlice`?

Reply:
Argh yes that should be `previouslyViewedObjectsSlice`. Will fix it now.